src/Workspaces/Core/Portable/Workspace/Solution/SolutionCompilationState.CompilationTracker.CompilationTrackerState.cs

```diff
@@ -205,6 +208,14 @@ private sealed class FinalCompilationTrackerState : CompilationTrackerState
                         unrootedSymbolSet);
                 }
 
+                public FinalCompilationTrackerState WithIsFrozen()
+                    => new(isFrozen: true,
```

Comment:
>new [](http://example.com/codeflow?start=23&length=3)

no check to return this if already frozen?

Reply:
fair.  i can do that.

Reply:
put into: https://github.com/dotnet/roslyn/pull/72111